src/teacher/views.py

```diff
@@ -1,113 +1,127 @@
+from django.views.generic import TemplateView, ListView, UpdateView, CreateView, DeleteView, DetailView
 
 from teacher.forms import TeacherAddForm, TeacherEditForm
+from teacher.mixins import TeacherValidFormMixin
 from .models import Teacher
 
 
-def index(request):
+class IndexTeacherDetailView(DetailView):
+    model = Teacher
+    template_name = 'teacher/index.html'
+    context_object_name = 'teacher'
 
-    id_teacher = request.GET.get('id')
-    if id_teacher is not None:
-        if not id_teacher.isdigit():
-            return HttpResponse("'id' must be numeric and greatest when zero")
+    def get_object(self, queryset=None):
 
-        id_teacher = int(id_teacher)
-        if id_teacher == 0:
-            return HttpResponse("'id' must be greatest when zero")
+        request = self.request
 
+        id_teacher = request.GET.get('id')
         try:
-            teacher = Teacher.objects.get(pk=id_teacher)
-        except Teacher.DoesNotExist:
-            return HttpResponse("Generate teachers in database using url: <strong>generate?teach_number=1</strong>" +
-                                "<br>" +
-                                "where <strong>teach_number</strong> is number of teachers. <br>Default: 100 teachers",
-                                status=200)
-        return render(request, "teacher/index.html", context={"teacher": teacher})
-    return HttpResponse("Введите параметр id: ?id=n")
+            group_obj = self.model.objects.get(pk=id_teacher)
+        except self.model.DoesNotExist:
+            raise Http404("Group does not exist! "
+                          "Generate teachers in database using url: generate?teach_number=1" +
+                          "where teach_number is number of teachers. Default: 100 teachers")
+        return group_obj
 
+    def get(self, request, *args, **kwargs):
 
-def gen_teacher(request):
+        id_teacher = request.GET.get('id')
+        if id_teacher is None:
+            return HttpResponse("Введите параметр id: ?id=n")
 
-    TEACHER_NUMBER = 100
+        if id_teacher is not None:
+            if not id_teacher.isdigit():
+                return HttpResponse("'id' must be numeric and greatest when zero")
 
-    teacher_number = str(request.GET.get('teach_number', TEACHER_NUMBER))
-    if teacher_number.isdigit():
-        try:
-            for _ in range(int(teacher_number)):
-                Teacher.generate_teacher()
-        except Exception as ex:
-            return HttpResponse(f'Data added fail! {ex}', status=500)
-        return redirect('teacher:data_success')
-    return HttpResponse("Error. The value must be numeric and greater than zero.", status=400)
+            id_teacher = int(id_teacher)
+            if id_teacher == 0:
+                return HttpResponse("'id' must be greatest when zero")
+
+        self.object = self.get_object()
+        context = self.get_context_data()
+        return self.render_to_response(context)
```

Comment:
it's better to call parent method:
```
        self.object = self.get_object()
        context = self.get_context_data()
        return self.render_to_response(context)
```
-->

```
        return super().get(request, *args, **kwargs)
```

Reply:
Fixed!